ted2zim/multi/scraper.py

```diff
@@ -61,57 +60,50 @@ def compute_format(item, fmt):
         return fmt.format(identity=item.replace(" ", "_"))
 
     @staticmethod
-    def download_playlist_list_from_site(topic_list):
+    def download_playlists_list_from_site(topics_list):
         records = []
-        for topic in topic_list:
+        for topic in topics_list:
             logger.debug(f"Getting playlists related to {topic}")
             playlist_sub_list = json.loads(
                 download_link(
                     f"https://www.ted.com/playlists/browse.json?topics={topic['value'].replace(' ', '+')}"
                 ).text
             )
             for record in playlist_sub_list["records"]:
-                if not [rec for rec in records if rec["id"] == record["id"]]:
+                if record not in records:
                     records.append(record)
             logger.debug("..OK")
         return records
 
-    def download_playlist_list_from_cache(self, s3_storage):
-        key = "kiwixfiles/playlist"
-        fpath = self.output_dir.joinpath("playlist_list.json")
-        if not s3_storage.has_object(key, s3_storage.bucket_name):
+    def download_playlists_list_from_cache(self, key, s3_storage):
+        fpath = self.output_dir.joinpath("playlists_list.json")
+        if not s3_storage.has_object(key):
             return False
         try:
-            (meta,) = s3_storage.get_object_head(key, only=[s3_storage.META_KEY])
-            if (
-                datetime.datetime.strptime(meta.get("expires"), "%Y-%m-%d %H:%M:%S.%f")
-                > datetime.datetime.now()
-            ):
+            meta = s3_storage.get_object_stat(key).meta
+            if datetime.date.fromisoformat(
+                meta.get("retrieved_on")
+            ) > datetime.date.today() - datetime.timedelta(days=7):
                 s3_storage.download_file(key, fpath)
             else:
-                raise Exception("File too old")
+                logger.error("File is too old")
```

Comment:
Shouldn't this be a warning? What file is too old? If referring to a previous log, use indentation to visually associate it for the user.

Reply:
Clarified that